fix: development with Docker

tancnle opened the pull request:
This commit uses the latest version of Centos 7 and Ruby 2.7 base image for Docker. The old image has been discontinued and caused breakage on build.

<details>
<summary>Error when running docker build</summary>

```sh
❯ docker build --tag headerdoc -f Dockerfile .
[+] Building 4.6s (8/15)                                                                          
 => [internal] load build definition from Dockerfile                                         0.0s
 => => transferring dockerfile: 640B                                                         0.0s
 => [internal] load .dockerignore                                                            0.0s
 => => transferring context: 2B                                                              0.0s
 => [internal] load metadata for docker.io/library/centos:centos8                            3.0s
 => [auth] library/centos:pull token for registry-1.docker.io                                0.0s
 => [internal] load build context                                                            0.0s
 => => transferring context: 50.43kB                                                         0.0s
 => [ 1/10] FROM docker.io/library/centos:centos8@sha256:a27fd8080b517143cbbbab9dfb7c8571c4  0.0s
 => CACHED [ 2/10] WORKDIR /headerdoc_build                                                  0.0s
 => ERROR [ 3/10] RUN dnf update -y && dnf install -y epel-release && dnf groupinstall -y "  1.5s
------
 > [ 3/10] RUN dnf update -y && dnf install -y epel-release && dnf groupinstall -y "Development Tools" && dnf install -y libxml2-devel perl-HTML-Parser perl-libwww-perl perl-FreezeThaw libxml2-devel wget perl-Devel-Peek ruby-devel rubygem-bundler && dnf clean all:
#7 1.457 CentOS Linux 8 - AppStream                       59  B/s |  38  B     00:00    
#7 1.463 Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
------
executor failed running [/bin/sh -c dnf update -y && dnf install -y epel-release && dnf groupinstall -y "Development Tools" && dnf install -y libxml2-devel perl-HTML-Parser perl-libwww-perl perl-FreezeThaw libxml2-devel wget perl-Devel-Peek ruby-devel rubygem-bundler && dnf clean all]: exit code: 1
```
</details>

## How to test

- Check out the branch
- Make sure Docker Desktop is installed
- Build a new Docker image
  ```sh
  docker build --tag headerdoc -f Dockerfile .
  ```
- Run the translator using the above image
  ```sh
  docker run --rm -v /Users/tanle/Code/learn/thoth-tech/splashkit-core:/splashkit/ headerdoc ./translate -i /splashkit/ -o /splashkit/generated -g cpp,docs,clib,python,pascal,csharp

  Executing cpp translator...                                     
  Done!                                                           
  Output written!                                                 

  Executing docs translator...                                    
  Done!                                                           
  Output written!                                                 
  Place `api.json` in the `data` directory of the `splashkit.io` repo                                                             

  Executing clib translator...                                    
  Done!                                                           
  Output written!                                                 

  Executing python translator...                                  
  Done!                                                           
  Output written!                                                 

  Executing pascal translator...                                  
  Done!                                                           
  Output written!                                                 

  Executing csharp translator...                                  
  Done!                                                           
  Output written!    
  ```
Tested on macOS 12.3.1 (Monterey) and Docker version 20.10.13, build a224086

tancnle committed:
fix: update Centos version for Docker

This commit uses the latest version of Centos 7 and Ruby 2.7 base image
for Docker. The old image has been discontinued and caused breakage on
build.

tancnle commented on Dockerfile:
```diff
@@ -23,5 +22,7 @@ RUN make realinstall
 
 COPY . /translator
 WORKDIR /translator
-RUN gem install bundler && bundle install --system
```
**thought:** Ruby 2.7 already comes with `bundle`. We do not need to install it again.

tancnle commented on Dockerfile:
```diff
@@ -1,20 +1,19 @@
-libxml2-devel \
-wget \
-perl-Devel-Peek \
-ruby-devel \
-rubygem-bundler \
```
**thought:** These 2 packages are already included in the `centos/ruby-27-centos7` base image.

tancnle commented:
👋🏼 @macite I would love to seek your expertise on this PR, please 🙏🏼

tancnle committed:
fix: remove duplicate dependencies

jacksievers commented:
![MicrosoftTeams-image](https://user-images.githubusercontent.com/101788313/161924075-33536bd3-527e-49c0-b20c-b88ca22aa4bb.png)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/101788313/161924121-0eb2e4e1-03d5-4cde-b357-175e155b8a3d.png)

Can confirm that these new changes work on Ubuntu 20.04 LTS

macite commented:
Thanks @tancnle. Could you add a compose file to simplify this? This could then have the command and volume all setup within it.

jacksievers reviewed (approved):
Successfully works on Ubuntu 20.04

dylanmedlin commented:
Tested on macOS Catalina 10.15.7
![Screen Shot 2022-04-08 at 12 47 29 am](https://user-images.githubusercontent.com/1919022/162228896-8ed0df0c-c5dd-4656-9ba7-209d77ba91d1.png)

allanfarrell committed:
docs: Update docker documentation

Update the docker setup instructions to include all steps required to get working with the repository. Included command line installation of docker for Ubuntu. These changes are primarily aimed at developers but can be used by all users for translator.

tancnle commented:
:wave: @allanfarrell I'm curious of why this PR is close. Is it intentional?

dylanmedlin commented:
I believe this PR should have been merged

tancnle commented:
Thanks, @dylanmedlin. 🙏🏼 

@macite I will open a separate PR to simplify the command using docker compose.
